src/main/java/com/robertx22/age_of_exile/gui/texts/textblocks/gearblocks/GearStatBlock.java

```diff
@@ -0,0 +1,209 @@
+
+    @Nonnull
+    private final GearItemData gearItemData;
+    @Nonnull
+    private final TooltipInfo info;
+
+    public BaseStatsData baseStatsData;
+
+    public ImplicitStatsData implicitStatsData;
+
+    public GearAffixesData gearAffixesData;
+
+    @Nullable
+    public UniqueStatsData uniqueStatsData;
+
+    public GearSocketsData gearSocketsData;
+
+    @Nullable
+    public GearEnchantData gearEnchantData;
+
+    private static int getPriority(String s, Map<String, Integer> priorityMap) {
+        return priorityMap.entrySet().stream()
+                .filter(entry -> s.contains(entry.getKey()))
+                .map(Map.Entry::getValue)
+                .min(Integer::compare)
+                .orElse(Integer.MAX_VALUE);
+    }
+
+    @Override
+    public List<? extends Component> getAvailableComponents() {
+        IgnoreNullList<Component> list = new IgnoreNullList<>();
+        MutableComponent emptyLine = ExileText.emptyLine().get();
+
+        this.implicitStatsData = gearItemData.imp;
+        this.baseStatsData = gearItemData.baseStats;
+        this.gearAffixesData = gearItemData.affixes;
+        this.gearSocketsData = gearItemData.sockets;
+        this.gearEnchantData = gearItemData.ench;
+        this.uniqueStatsData = gearItemData.uniqueStats;
+
+        if (baseStatsData != null) {
+            list.addAll(baseStatsData.GetTooltipString(info, gearItemData));
+            list.add(emptyLine);
+        }
+
+        boolean showMerge = !info.useInDepthStats();
+        if (showMerge) {
+
+            List<ExactStatData> stats = new ArrayList<>();
+
+            stats.addAll(implicitStatsData.GetAllStats(gearItemData));
+
+            gearAffixesData.getAllAffixesAndSockets().forEach(x -> stats.addAll(x.GetAllStats(gearItemData)));
+
+            if (uniqueStatsData != null) {
+                stats.addAll(uniqueStatsData.GetAllStats(gearItemData));
+            }
+
+            Map<Boolean, List<ExactStatData>> map = stats.stream().collect(Collectors.groupingBy(x -> x.getStat().is_long));
+
+            MergedStats merged = new MergedStats(new ArrayList<>(), info);
+            if (map.get(false) != null) {
+                merged = new MergedStats(map.get(false), info);
+            }
+
+            HashMap<String, Integer> DPMap = new HashMap<>();
+            DPMap.put("damage", 1);
+            DPMap.put("dmg", 1);
+            DPMap.put("penetration", 2);
+
+            Comparator<TooltipStatInfo> damageAndPenetration = (s1, s2) -> {
+                int p1 = getPriority(s1.stat.GUID(), DPMap);
+                int p2 = getPriority(s2.stat.GUID(), DPMap);
+                return Integer.compare(p1, p2);
+            };
+
+            Map<String, Integer> priorityMap = new HashMap<>();
+            priorityMap.put("fire", 2);
+            priorityMap.put("lightning", 5);
+            priorityMap.put("cold", 4);
+            priorityMap.put("chaos", 3);
+            priorityMap.put("physical", 1);
```

Comment:
Why isn't this priority in the element enum

Reply:
this order is from Stat hud.
![image](https://github.com/RobertSkalko/Mine-And-Slash-Rework/assets/21279092/541c5204-c8b4-475c-9e99-747b586fa98b)
I'll use the order from Elements, but why the ALL element doesn't have a `dmgname`? I don't think putting an empty String into the order map is a good idea.